File: app/src/main/java/com/example/mrakopediareader/categorieslist/AllCategories.java
package com.example.mrakopediareader.categorieslist;

import com.example.mrakopediareader.api.dto.Category;

import java.util.ArrayList;

import io.reactivex.rxjava3.core.Observable;

public class AllCategories extends CategoriesList {
    @Override
    protected Observable<ArrayList<Category>> getCategories() {
        assert this.api != null;
        return this.api.getCategories();
    }
}

File: app/src/main/java/com/example/mrakopediareader/categorieslist/AllCategories.kt
package com.example.mrakopediareader.categorieslist

import com.example.mrakopediareader.api.dto.Category
import io.reactivex.rxjava3.core.Observable

class AllCategories : CategoriesList() {
    override fun getCategories(): Observable<List<Category>> {
        return api.getCategories()
    }
}